orte/runtime/orte_data_server.c

```diff
@@ -178,7 +178,7 @@ void orte_data_server(int status, orte_process_name_t* sender,
     uint32_t uid = UINT32_MAX;
     opal_pmix_data_range_t range;
     orte_data_req_t *req, *rqnext;
-    orte_jobid_t jobid;
+    orte_jobid_t jobid = 0;
```

Comment:
I guess it must think there is some path that allows jobid to be used without having been unpacked? I can't readily see it - would be nice to know what it is and fix it.

Regardless, you'd be better off initializing it to ORTE_JOBID_INVALID as job=0 is the daemon job and a legitimate value.